WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentsFragment.kt

```diff
@@ -78,11 +82,28 @@ class SiteCreationIntentsFragment : Fragment() {
     private fun SiteCreationIntentsFragmentBinding.updateUiState(uiState: IntentsUiState) {
         (recyclerView.adapter as SiteCreationIntentsAdapter).update(uiState.content.items)
         updateTitleVisibility(uiState.isAppBarTitleVisible)
-        if (!uiState.isHeaderVisible) input.requestFocus()
+
+        uiState.vertical
+                ?.let {
+                    input.requestFocus()
+                    ActivityUtils.showKeyboard(input)
+                }
+                ?: run {
+                    if (!uiState.searchQuery.isNullOrBlank()) return@run
+                    input.setTextUnobserved("")
+                    ActivityUtils.hideKeyboard(requireActivity())
+                }
+
         siteCreationIntentsHeader.root.isVisible = uiState.isHeaderVisible
         recyclerView.scrollToPosition(0)
     }
 
+    private fun TextInputEditText.setTextUnobserved(text: CharSequence) {
+        removeTextChangedListener(inputWatcher)
+        setText(text)
+        addTextChangedListener(inputWatcher)
```

Comment:
I _think_ this is safe, since I guess `setTextUnobserved` would only ever run _after_ the `onViewStateRestored`, but maybe we can add a `null` check just in case, to guard unexpected races where `inputWatcher` is `null` here? Wdyt?

Reply:
I wrapped the code in a `?.let {}` block for this in [3642566](https://github.com/wordpress-mobile/WordPress-Android/pull/16380/commits/364256618ca387ad01578987a24f31cad202bfd1), thank you for this remark also, I agree safer is better 🎉 .